internal/controller/management/project-configs/project_configs.go

```diff
@@ -0,0 +1,258 @@
+		return ctrl.Result{}, reconcileErr
+	}
+	// Immediate requeue if needed.
+	if needsRequeue {
+		return ctrl.Result{Requeue: true}, nil
+	}
+	// Otherwise, requeue after a delay.
+	// TODO: Make the requeue delay configurable.
+	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
+}
+
+func (r *reconciler) syncProjectConfig(
+	ctx context.Context,
+	projectConfig *kargoapi.ProjectConfig,
+) (kargoapi.ProjectConfigStatus, bool, error) {
+	logger := logging.LoggerFromContext(ctx)
+	status := projectConfig.Status.DeepCopy()
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReconciling,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionFalse,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+
+	whReceivers, err := r.ensureWebhookReceiversFn(ctx, projectConfig)
+	if err != nil {
+		logger.Error(err, "error ensuring webhook receivers")
+		conditions.Set(status, &metav1.Condition{
+			Type:               kargoapi.ConditionTypeReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "Syncing",
+			Message:            "Failed to ensure project config webhook receivers: " + err.Error(),
+			ObservedGeneration: projectConfig.GetGeneration(),
+		})
+		return *status, true, fmt.Errorf("error ensuring webhook receivers: %w", err)
+	}
+	status.WebhookReceivers = whReceivers
+
+	conditions.Delete(status, kargoapi.ConditionTypeReconciling)
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Synced",
+		Message:            "ProjectConfig is synced and ready for use",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+	return *status, false, nil
+}
+
+func (r *reconciler) ensureWebhookReceivers(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) ([]kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	if pc.Spec.WebhookReceiverConfigs == nil {
+		logger.Debug("ProjectConfig does not have any receiver configurations")
+		return nil, nil
+	}
+
+	logger.Debug("ensuring receivers",
+		"receiver-configs", len(pc.Spec.WebhookReceiverConfigs),
+	)
+
+	var webhookReceivers []kargoapi.WebhookReceiver
+	for _, rc := range pc.Spec.WebhookReceiverConfigs {
+		if rc.GitHub != nil {
+			whr, err := r.ensureGitHubWebhookReceiver(ctx, pc, rc)
+			if err != nil {
+				logger.Error(err, "error ensuring GitHub webhook receiver",
+					"receiver-config", rc,
+				)
+				return nil, fmt.Errorf("error ensuring GitHub webhook receiver: %w", err)
+			}
```

Comment:
This worries me because if we fail processing the config for _one_ receiver (say the Secret is missing or malformed) we'll end up with _no_ receivers.

It might be better to set some kind of "degraded" condition on the config instead and then carry on.

Reply:
I've pushed https://github.com/akuity/kargo/pull/4000/commits/edc8bc60bffa26a382774c45eb4594fbbfad96a5.

Let me know if this aligns with what you had in mind.

I don't think I like the change though. For example:

> This worries me because if we fail processing the config for one receiver (say the Secret is missing or malformed) we'll end up with no receivers.

Seems reasonable to me. If we end up with less receivers than what we designated that doesn't feel like we have really reconciled so I feel like failing holistically is the behavior I would expect.

cc @hiddeco

Reply:
I would set `Ready=False` but persist as many webhooks as possible to the status. `Ready=False` already means "degraded" in Kubernetes API terms, as things are not as we expect them to be. A separate condition would only spark confusion, especially for a "config".